src/hotspot/cpu/aarch64/aarch64_neon_ad.m4

```diff
@@ -133,6 +133,72 @@ instruct reinterpretX2D(vecD dst, vecX src)
   ins_pipe(vlogical64);
 %}
 
+instruct reinterpretS2X(vecX dst, vecD src)
+%{
+  predicate(n->bottom_type()->is_vect()->length_in_bytes() == 16 &&
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes() == 4);
+  match(Set dst (VectorReinterpret src));
+  ins_cost(INSN_COST);
+  format %{ " # reinterpret $dst,$src\t# S2X" %}
+  ins_encode %{
+    // If registers are the same, no register move is required - the
+    // upper 96 bits of 'src' are expected to have been initialized
+    // to zero.
+    if (as_FloatRegister($dst$$reg) != as_FloatRegister($src$$reg)) {
+      __ dups(as_FloatRegister($dst$$reg), __ T2S, as_FloatRegister($src$$reg));
+    }
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct reinterpretX2S(vecD dst, vecX src)
+%{
+  predicate(n->bottom_type()->is_vect()->length_in_bytes() == 4 &&
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes() == 16);
+  match(Set dst (VectorReinterpret src));
+  ins_cost(INSN_COST);
+  format %{ " # reinterpret $dst,$src\t# X2S" %}
+  ins_encode %{
+    // Resize the vector from 128-bits to 32-bits. The higher 96-bits of
+    // the "dst" register must be cleared to zero.
+    __ dups(as_FloatRegister($dst$$reg), __ T2S, as_FloatRegister($src$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct reinterpretS2D(vecD dst, vecD src)
+%{
+  predicate(n->bottom_type()->is_vect()->length_in_bytes() == 8 &&
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes() == 4);
+  match(Set dst (VectorReinterpret src));
+  ins_cost(INSN_COST);
+  format %{ " # reinterpret $dst,$src\t# S2D" %}
+  ins_encode %{
+    // If registers are the same, no register move is required - the
+    // upper 32 bits of 'src' are expected to have been initialized
+    // to zero.
+    if (as_FloatRegister($dst$$reg) != as_FloatRegister($src$$reg)) {
+      __ dups(as_FloatRegister($dst$$reg), __ T2S, as_FloatRegister($src$$reg));
+    }
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct reinterpretD2S(vecD dst, vecD src)
+%{
+  predicate(n->bottom_type()->is_vect()->length_in_bytes() == 4 &&
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes() == 8);
+  match(Set dst (VectorReinterpret src));
+  ins_cost(INSN_COST);
+  format %{ " # reinterpret $dst,$src\t# D2S" %}
+  ins_encode %{
+    // Resize the vector from 64-bits to 32-bits. The higher 32-bits of
+    // the "dst" register must be cleared to zero.
+    __ dups(as_FloatRegister($dst$$reg), __ T2S, as_FloatRegister($src$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
```

Comment:
I'm sure you can find a more brief way to write these four patterns. Please do.

Reply:
OK, refactored.

Reply:
Please do so for the others. Thanks.

Reply:
OK，I refactored the reinterpretX2D and reinterpretD2X rule.